index.js

```diff
@@ -75,6 +82,14 @@ function printArray(val, refs, opts, state) {
   return val.constructor.name + ' ' + printList(val, refs, opts, state);
 }
 
+function printPlugin(val, refs, opts, state) {
+  var plugin = plugins.find(function(plugin){
+    return plugin.test(val);
+  });
+
+  return plugin.print(val, refs, opts, state, print);
```

Comment:
I was thinking of running the plugin print function like this to keep internal stuff hidden:

```js
function printPlugin(val, refs, opts, state) {
  // ...

  const boundPrint = val => print(val, refs, opts, state);
  const boundIndent = val => indent(val, opts);

  return plugin.print(val, boundPrint, boundIndent);
}
```